src/main/java/com/metamx/emitter/core/ParametrizedUriEmitter.java

```diff
@@ -120,10 +121,30 @@ public void close() throws IOException
   @Override
   public void flush() throws IOException
   {
-    synchronized (emitters) {
-      for (Emitter emitter : emitters.values()) {
-        emitter.flush();
+    Exception thrown = null;
+    for (HttpPostEmitter httpPostEmitter : emitters.values()) {
+      try {
+        httpPostEmitter.flush();
       }
+      catch (Exception e) {
+        // If flush was interrupted, exit the loop
+        if (Thread.currentThread().isInterrupted()) {
+          if (thrown != null) {
+            e.addSuppressed(thrown);
+          }
+          throw Throwables.propagate(e);
```

Comment:
This method is deprecated since Guava 20.0, which also adds `Throwables.throwIfUnchecked()` which facilitates transition off `Throwables.propagate()`. Until using Guava 20.0 it's 4 lines of boilerplate code vs. 1 line so I think it's simpler to use the older method, until we update our dep to Guava 20.0